src/resource/search/dto/search.result.dto.ts

```diff
@@ -0,0 +1,36 @@
+import { EventEntity } from "src/database/entities/event.entity"
+import { NoteEntity } from "src/database/entities/notes.entity"
+import { ReturnTaskDTO } from "src/resource/task/dto";
+
+enum ResourceType {
+    NOTES = "NOTES",
+    TASKS = "TASKS",
+    EVENTS = "EVENTS"
+}
+
+type Entity = NoteEntity | ReturnTaskDTO | EventEntity;
+type Result = Record<ResourceType, Entity[]>
+
+export class SearchResult {
+    constructor() {
+        this.result = {
+            NOTES: [],
+            TASKS: [],
+            EVENTS: [],
+        }
+    }
+
+    result: Result
+
+    setNotes(notes: NoteEntity[]) {
+        this.result.NOTES = notes;
+    }
+
+    setTasks(tasks: ReturnTaskDTO[]) {
+        this.result.TASKS = tasks;
+    }
+
+    setEvents(events: EventEntity[]) {
+        this.result.EVENTS = events;
+    }
+}
```

Comment:
I think it is not necessary to use class in this case - you may ask why? - That is because only think it do is setting its properties. If  we have properly typed object we can just create it on the go like const result = { key1: .... etc. }